tests/melee_dodge_hit_test.cpp

```diff
@@ -56,8 +56,18 @@ static float dodge_wearing_item( avatar &dummy, item &clothing )
 {
     // Get nekkid and wear just this one item
 
-    item_location loc = item_location( *dummy.as_character(), &dummy.i_at( -2 ) );
-    while( dummy.takeoff( loc ) ) {}
+    while( true ) {
+        item &it = dummy.i_at( -2 );
+
+        //If the item is a null pointer then we don't need to take it off.
```

Comment:
Strictly speaking this does not check if it's a null pointer, but checks if it's a special `null` item. The function itself should never return a null reference.

Reply:
Aw that makes sense. That's funny that even when I take your solution I still add error in the form of a comment!

Thanks for looking at this with me.